### DISCUSSION TITLE
Redirect Issue

### DISCUSSION THREAD

**Discussion by mikeeyo23** [codeigniter4/shield#978] [Q&A]
Hello all,

First, thanks to everyone for your help over the past year. I previously has a completely working production installation with zero issues- I had some issues on host side that prompted me to essentially delete the website and re-install from scratch. I have read the beta 7 and beta 8 guides (i have tried both versions with the same issue), and have confirmed all the settings that should be enacted have been. 

Also, http/https doesn't result in a different behavior in this regard- nor does public vs non-public.

Now to the actual issue:

I have added the  `'session' => ['except' => ['login*', 'register', 'auth/a/*', 'logout']]`, code 
I can register my user with no problem, login works flawlessly. However, `/logout` (which I've also confirmed the routes of) redirects back to the protected homepage and not the login page. I can then use `/login` to redirect back to the login page which then shows the expected "You have been successfully logged out" txt. But, register, magiclink, and login actions instantly take me back to the protected homepage as if it never killed the login session.

CI isn't generating any errors and the rest of CI appears to be operating 100% as expected, is there anything I can check? I'd be happy to add a user account for someone to sign in with to help solve the problems as needed. 

EDIT: if I attempt to re-login after manually redirecting to `/login` I receive the following error:

CodeIgniter\Shield\Exceptions\LogicException
> The user has User Info in Session, so already logged in or in pending login state. If a logged in user logs in again with other account, the session data of the previous user will be used as the new user. Fix your code to prevent users from logging in without logging out or delete the session data. user_id: 1

**Comment by kenjis** [comment]
Check `app/Config/Auth.php` if it is the same as the latest one.
https://github.com/codeigniter4/shield/blob/develop/src/Config/Auth.php

**Reply by mikeeyo23** [reply]
@kenjis,

Thank you for your response. To confirm I copied / pasted that file and replaced my current file. No change in behavior.

**Reply by kenjis** [reply]
It seems you did not logout. But I don't know why.
Can you show the routing table with `php spark routes`?

**Comment by kairiex2** [comment]
@mikeeyo23 Clear your web related cookie from browser. Or better if you know the session id related to that user then just delete the session from your server. I encountered this issue too when shield redirecting after switching from http to https or vice versa. It seems conflict happens when i set base_url on .env but when typing url to access page not using the same from it (set http but type https or the other way) or when i use force https on already log in user.

> The user has User Info in Session, so already logged in or in pending login state. If a logged in user logs in again with other account, the session data of the previous user will be used as the new user. Fix your code to prevent users from logging in without logging out or delete the session data. user_id: 1
